keyring.php

```diff
@@ -325,6 +329,75 @@ static function token_select_box( $tokens, $name, $create = false ) {
 	static function is_error( $obj ) {
 		return is_a( $obj, 'Keyring_Error' );
 	}
+
+	/**
+	 * For some services the request params are serialised into the state param. This
+	 * method unserialises these and adds them back to the _REQUEST global
+	 *
+	 * @param string $state The serialised $_REQUEST['state'] param
+	 * @return void
+	 */
+	static function unpack_state_parameters( $state ) {
+		$unpacked_state =  unserialize( base64_decode( $state ) );
```

Comment:
Have you verified what happens here when this tries to run against something that's _not_ packed params? e.g. for any other service where it's expecting just an int that points to a token in the DB?

Reply:
Nothing happens for existing state as the main logic block of this method is skipped if unserialize doesn't result in an array with a 'hash' property. Have tested this on sandbox and all a sample of other services, eg. facebook, twitter, linked in could all be added added and removed as normal.